planner/mapper/mapper.go

```diff
@@ -797,50 +803,66 @@ func resolveInnerFilterDependencies(
 	source map[string]any,
 	mapping *core.DocumentMapping,
 	existingFields []Requestable,
+	resolvedFields []Requestable,
 ) ([]Requestable, error) {
 	newFields := []Requestable{}
 
-sourceLoop:
 	for key := range source {
-		if strings.HasPrefix(key, "_") && key != request.KeyFieldName {
+		if key == opAnd || key == opOr { // handle _not
+			andFilter := source[key].([]any)
```

Comment:
nitpick: `andFilter` should be `andOrFilter` (or something else) no?  It is not just an and-filter?